### PR TITLE
Homework4: Anton Kochnevski

### PR DESCRIPTION
**Pull request by Antonkoch999**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/Homework4/Languages_special.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+Языки
+Каждый из N школьников некоторой школы знает Mi языков.
+Определите, какие языки знают все школьники и языки,
+которые знает хотя бы один из школьников.
+"""
+
+
+def students(n, m1, m2, m3):
```
Comment (belovmd) [inline src/Homework4/Languages_special.py]: unclear input. 
why do we need two solution? let only one be left

File: src/Homework4/Languages_general.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        lst = []
+        for language in range(count):
+            lst.append(str(input('Введите иностранные языки:')))
+            dct[mi] = lst
+
+    list_values = list(dct.values())
+    all_language = []
+    for el in list_values:
+        for lang in el:
+            if lang in all_language:
```
Comment (belovmd) [inline src/Homework4/Languages_general.py]: need to revert condition

File: src/Homework4/Languages_general.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            else:
+                all_language.append(lang)
+
+    all_know = []
+    all_know_number = 0
+    for elem in all_language:
+        for key, value in dct.items():
+            if elem not in value:
+                break
+            elif elem in all_know:
```
Comment (belovmd) [inline src/Homework4/Languages_general.py]: revert condition

File: src/Homework4/Languages_general.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                continue
+            else:
+                all_know_number += 1
+                all_know.append(elem)
+
+    one_know = []
+    one_know_number = 0
+    for elem in all_language:
+        for key, value in dct.items():
+            if elem in one_know:
```
Comment (belovmd) [inline src/Homework4/Languages_general.py]: bad practise

File: src/Homework4/Loor_back.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    if a < b:
+        a, b = b, a
+    while a > 0:
+        if b <= 0:
+            break
+        a = a % b
+        if a <= 0:
+            break
+        b = b % a
+    if a > b:
```
Comment (belovmd) [inline src/Homework4/Loor_back.py]: just a hack because you don't know where is solution. 
can be made more clear and simple

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework4.md):
- 12
- 15
- 17
- 18
- 22

File: src/Homework4/Languages_special.py (hunk 1 above)
Comment (Antonkoch999) [reply src/Homework4/Languages_special.py]: deleted the file

File: src/Homework4/Loor_back.py (hunk 5 above)
Comment (Antonkoch999) [reply src/Homework4/Loor_back.py]: corrected

File: src/Homework4/Languages_general.py (hunk 4 above)
Comment (Antonkoch999) [reply src/Homework4/Languages_general.py]: corrected

File: src/Homework4/Languages_general.py (hunk 3 above)
Comment (Antonkoch999) [reply src/Homework4/Languages_general.py]: corrected

File: src/Homework4/Languages_general.py (hunk 2 above)
Comment (Antonkoch999) [reply src/Homework4/Languages_general.py]: corrected

File: src/Homework4/Loor_back.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def evklid(number1, number2):
+
+    while number1 and number2:
+        if number1 > number2:
+            number1 = number1 % number2
+        else:
+            number2 = number2 % number1
+    return number1 + number2
```
Comment (belovmd) [inline src/Homework4/Loor_back.py:16]: it's also just a hack )))

**Review by belovmd — COMMENTED** [review]
9

File: src/Homework4/Loor_back.py (hunk 6 above)
Comment (Antonkoch999) [reply src/Homework4/Loor_back.py:16]: This is the first solution on the internet)

**Pull request closed without merging**